packages/svelte/src/compiler/phases/2-analyze/visitors/RegularElement.js

```diff
@@ -20,11 +20,11 @@ import { mark_subtree_dynamic } from './shared/fragment.js';
 export function RegularElement(node, context) {
 	validate_element(node, context);
 
-	check_element(node, context.state);
+	check_element(node, context);
 
 	node.metadata.path = [...context.path];
 
-	context.state.analysis.elements.push(node);
+	context.state.analysis.elements.push({ node, path: context.path });
```

Comment:
We already have `path` as `metadata` on the node, would be better to either reuse that or to replace usages of `metadata.path` where possible.
(having `metadata.path` might make it easier for the CSS prune stuff)

Reply:
As far as I can see we're using `metadata.path` in only one place. Seems like we could probably get rid of it. Using `metadata.path` in pruning is a possibility but it seems easier to just use `context.path` everywhere instead

Reply:
Actually I take that back, I misunderstood — reverting most recent change

Reply:
> but it seems easier to just use context.path everywhere instead

But during pruning context.path will relate to the css tree and not the element's parent tree

Reply:
I meant the `context.path` being stored on `analysis.elements`. But on reflection it does make sense to use `metadata.path` everywhere — you probably missed that reply because GitHub's UI is a piece of shit